From 9fffd69ce88153a2becb81fe4e6083aa3b57dfd4 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Fri, 5 Sep 2014 01:51:41 -0400
Subject: [iommu] vt-d: returning free pointer in get_domain_for_dev()

Message-id: <20140905015140.31297.79168.stgit@gir.stowe>
Patchwork-id: 91352
O-Subject: [RHEL7.1 PATCH 59/65] iommu/vt-d: returning free pointer in get_domain_for_dev()
Bugzilla: 1129880 1087643
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1087643
Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1129880
Upstream Status: 14d405699634d4ce0adfc7b4f52ac7427220a98d

commit 14d405699634d4ce0adfc7b4f52ac7427220a98d
Author: Dan Carpenter <dan.carpenter@oracle.com>
Date:   Fri Mar 28 11:29:50 2014 +0300

    iommu/vt-d: returning free pointer in get_domain_for_dev()

    If we hit this error condition then we want to return a NULL pointer and
    not a freed variable.

    Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
    Signed-off-by: David Woodhouse <David.Woodhouse@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/intel-iommu.c b/drivers/iommu/intel-iommu.c
index 6fbce01..69fa7da 100644
--- a/drivers/iommu/intel-iommu.c
+++ b/drivers/iommu/intel-iommu.c
@@ -2257,6 +2257,7 @@ static struct dmar_domain *get_domain_for_dev(struct device *dev, int gaw)
   goto error;
  if (iommu_attach_domain(domain, iommu)) {
   free_domain_mem(domain);
+  domain = NULL;
   goto error;
  }
  free = domain;
-- 
1.7.1